### ISSUE TITLE
Bolt adapter version not implemented exception raised on db migration

### ISSUE THREAD

**Issue by gottfrois** [neo4jrb/neo4j-core#303]
I'm getting the following error when executing `bundle exec rake neo4j:migrate`:

```
rake aborted!
should be implemented!
/Users/gottfrois/.rvm/gems/ruby-2.4.2/gems/neo4j-core-8.0.0/lib/neo4j/core/cypher_session/adaptors/bolt.rb:57:in `version'
/Users/gottfrois/.rvm/gems/ruby-2.4.2/gems/neo4j-core-8.0.0/lib/neo4j/core/cypher_session.rb:22:in `block (2 levels) in <class:CypherSession>'
/Users/gottfrois/.rvm/gems/ruby-2.4.2/gems/neo4j-9.0.5/lib/neo4j/tasks/migration.rake:67:in `check_neo4j_version_3'
/Users/gottfrois/.rvm/gems/ruby-2.4.2/gems/neo4j-9.0.5/lib/neo4j/tasks/migration.rake:76:in `block (3 levels) in <top (required)>'
/Users/gottfrois/.rvm/gems/ruby-2.4.2/gems/neo4j-9.0.5/lib/neo4j/tasks/migration.rake:118:in `block (3 levels) in <top (required)>'
/Users/gottfrois/.rvm/gems/ruby-2.4.2/gems/neo4j-9.0.5/lib/neo4j/tasks/migration.rake:45:in `block (2 levels) in <top (required)>'
/Users/gottfrois/.rvm/gems/ruby-2.4.2/gems/rake-12.1.0/exe/rake:27:in `<top (required)>'
/Users/gottfrois/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `eval'
/Users/gottfrois/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `<main>'
```

Looking at the `bolt.rb` file there is indeed an exception raised here:
https://github.com/neo4jrb/neo4j-core/blob/v8.0.0/lib/neo4j/core/cypher_session/adaptors/bolt.rb#L57

### Runtime information:

Neo4j database version: `3.2`
`neo4j` gem version: `9.0.5`
`neo4j-core` gem version: `8.0.0`

**Comment by cheerfulstoic** [comment]
;(  Sorry, I'm not sure why that didn't get implemented.

Because there are one or two kinks with the Bolt adaptor and because I haven't had time to dig into them, I've unfortunately been needing to recommend the HTTP adaptor for now.

Thank you so much for this issue!

**Comment by JustinAiken** [comment]
> I've unfortunately been needing to recommend the HTTP adaptor for now.

Just for migrations, or in general?

**Comment by cheerfulstoic** [comment]
Hey, I did a bit of work and found a way to fix this.  I'm heading to bed shortly so I don't want to push it out right now, but I should be able to push it out tomorrow

**Comment by cheerfulstoic** [comment]
Regarding Bolt, I meant generally.  Though I hope to have the Bolt in better shape in the near future.  Honestly it largely works, there are just a few edge cases like this one.  Also, in benchmarking it it's generally at least somewhat slower than the HTTP implementation.  It might be good to use the [neo4j-client](https://neo4j-client.net/) library under the covers since the various big DB gems (for postgres, mysql, and sqlite) all have C implementations bound underneath

**Comment by cheerfulstoic** [comment]
Sorry, I didn't get a chance to finish this up today.  It's definitely next on my list and I should be able to finish it by early next week

**Comment by cheerfulstoic** [comment]
This should now be fixed in version 8.0.3 of the `neo4j-core` gem.  Give it a try!
